strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/AuthenticatorsClassLoader.java

```diff
@@ -0,0 +1,76 @@
+package org.carlspring.strongbox.authentication.registry.support;
+
+import org.carlspring.strongbox.resource.ConfigurationResourceResolver;
+import org.carlspring.strongbox.util.JarFileClassLoader;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.stream.Stream;
+
+import com.google.common.base.Throwables;
+import org.apache.commons.lang.ArrayUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+public class AuthenticatorsClassLoader
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(AuthenticatorsClassLoader.class);
+
+    public static void loadAuthenticatorsClasses()
+    {
+
+        File authenticatorsDirectory = null;
+        try
+        {
+            authenticatorsDirectory = ConfigurationResourceResolver.getConfigurationResource("lib.authentication",
```

Comment:
Your requirements from SB-625: 
> It should be possible to configure this during run-time
 > It should be possible to reload the configuration without having to restart the server

Here is the general idea behind my solution:

1. Initially, the authentication providers are listed in the etc/conf/strongbox-authentication-providers.xml file
- in the strongbox-authentication-providers.xml file, there may be listed both spring injected components and custom java objects
- in the strongbox-authentication-providers.xml file, there may be listed supported providers or user defined providers
- user defined providers may be scanned in initially in the delivered jar files inside the authentication-lib directory
2. In the runtime, admins are able to reload the authentication providers
- etc/conf/strongbox-authentication-providers.xml file may be edited in the runtime
- admins are able to add/remove/reorder authentication providers
- authentication-lib directory may be supplied by some new authentication providers inside delivered jar files 
- authentication providers mentioned in the etc/conf/strongbox-authentication-providers.xml don't have to be in the classpath - there is a flexibility to add them in the runtime to the authentication-lib directory

Here is a simple example of the authentication provider that is supported by the implemented authentication mechanism https://github.com/fuss86/strongboxauthenticationsample 

There is a big flexibility in creating new providers using provided API, I think